scripts/warpwire.js

```diff
@@ -0,0 +1,509 @@
+        return;
+      }
+
+      return player.getPlaybackRate();
+    };
+
+    /**
+     * Set current playback rate.
+     * Listen to event "playbackRateChange" to check if successful.
+     *
+     * @public
+     * @params {Number} suggested rate that may be rounded to supported values
+     */
+    self.setPlaybackRate = function (newPlaybackRate) {
+      if (!player || !player.setPlaybackRate) {
+        return;
+      }
+
+      playbackRate = Number(newPlaybackRate);
+      player.setPlaybackRate(playbackRate);
+    };
+
+    /**
+     * Set current captions track.
+     *
+     * @param {H5P.Video.LabelValue} Captions track to show during playback
+     */
+    self.setCaptionsTrack = function (track) {
+      if (!player || !player.setCaption) {
+        return;
+      }
+      if (track == null) {
+        player.setCaption('');
+      } else {
+        player.setCaption(track.value);
+      }
+    };
+
+    /**
+     * Figure out which captions track is currently used.
+     *
+     * @return {H5P.Video.LabelValue} Captions track
+     */
+    self.getCaptionsTrack = function () {
+      if (!player || !player.getCaptions) {
+        return;
+      }
+      trackList = player.getCaptions();
+       if (trackList.length > 0) {
+         for (var i = 0; i < trackList.length; i++) {
+           if (trackList[i].enabled == true) {
+            return (track.label ? new H5P.Video.LabelValue(track.label, track.label) : null);
+           }
+         }
+      }
+    };
+
+    // Respond to resize events by setting the YT player size.
+    self.on('resize', function () {
+      if (!$wrapper.is(':visible')) {
+        return;
+      }
+
+      if (!player) {
+        // Player isn't created yet. Try again.
+        create();
+        return;
+      }
+
+      if (!player || !player.getIframe) {
+        return;
+      }
+
+      // Use as much space as possible
+      $wrapper.css({
+        width: '100%',
+        height: '100%'
+      });
+
+      var width = $wrapper[0].clientWidth;
+      var height = options.fit ? $wrapper[0].clientHeight : (width * (9/16));
+      
+      // Validate height before setting
+      if (height > 0) {
+        // Set size
+        $wrapper.css({
+          width: width + 'px',
+          height: height + 'px'
+        });
+
+        iframe = player.getIframe();
+        iframe.width = width;
+        iframe.height = height;
+      }
+    });
+  }
+
+  /**
+   * Check to see if we can play any of the given sources.
+   *
+   * @public
+   * @static
+   * @param {Array} sources
+   * @returns {Boolean}
+   */
+  Warpwire.canPlay = function (sources) {
+    return getId(sources[0].path);
+  };
+
+  /**
+   * Find id of Warpwire video from given URL.
+   *
+   * @private
+   * @param {String} url
+   * @returns {String} Warpwire video identifier
+   */
+
+  var getId = function (url) {
+    const matches = url.match(/^[^\/]+:\/\/([^\/]*\.[^\/]+)\/w\/(.+)\/.*$/);
```

Comment:
Isn't this a bit generic? I.e: Won't warpwire.com always be part of the url?

Reply:
@fnoks Unfortunately, not. Our clients are allowed to use custom domains that CNAME to their basic domain name (which does end in warpwire.com). If you feel this URL really is too generic, I will see if we can force links to always use the basic domain name, though I'm not sure if this will be feasible in all cases.

We can also make the test a little tighter. URL paths will always start with `/w/<6 char code>/`, where the 6 char code is alphanumeric plus underscores and dashes. It can't be anything else. We could replace `\w` in the regex with `[-_0-9a-zA-Z]{6}`.

Reply:
It is a bit broad because Warpwire URLs can be whitelabeled. We can improve this regex slightly by changing the `\/w\/(.+)` to `\/w\/[-_0-9a-zA-Z]{6,7}`.

Reply:
@fnoks this has been fixed.